Final project num 11 done

RinatkaDyrov opened the pull request:
Added service logic and new functionality
All postman tests passed
Added unit tests

Commit:
VIP: sprint 11
Added storage interfaces, inMemory classes and services

RinatkaDyrov committed:
VIP: sprint 11
60/76 postman tests passed

RinatkaDyrov committed:
Final project num 11 done
Added service logic and new functionality
All postman tests passed
Added unit tests

RinatkaDyrov committed:
Final project num 11 done
Added service logic and new functionality
All postman tests passed
Added unit tests
Unused imports were deleted

VladimirTitov4 commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,76 +1,52 @@
+
+    private final FilmService filmService;
+
+    public FilmController(FilmStorage filmStorage, FilmService filmService) {
```
По поводу внедрение бина через конструктор - в целом все ок, но попробуйте использовать аннотацию @RequiredArgsContructor - достаточно поставить ее и явно конструктор не объявлять. 
Это полезно, когда внедряется еще несколько бинов в класс и нет необходимости их прописывать в конструкторе, просто делаете новое приватное финальное поле и все

VladimirTitov4 commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,76 +1,52 @@
+    }
+
+    @PutMapping("/{id}/like/{userId}")
+    public void setLike(@PathVariable("id") long id, @PathVariable("userId") long userId) {
```
В скобках ("id") не обязательно ставить, если имена параметров совпадают

VladimirTitov4 commented on src/main/java/ru/yandex/practicum/filmorate/handlers/ErrorHandler.java:
```diff
@@ -0,0 +1,31 @@
+import ru.yandex.practicum.filmorate.exception.ValidationException;
+
+@RestControllerAdvice
+public class ErrorHandler {
```
Отлично с обработкой ошибок!

VladimirTitov4 commented on src/main/java/ru/yandex/practicum/filmorate/storage/InMemoryFilmStorage.java:
```diff
@@ -0,0 +1,84 @@
+
+    @Override
+    public Collection<Film> findAll() {
+        return List.copyOf(films.values());
```
Правильно, что оборачиваете в отдельную коллекцию

VladimirTitov4 commented on src/main/java/ru/yandex/practicum/filmorate/storage/InMemoryUserStorage.java:
```diff
@@ -0,0 +1,112 @@
+
+    private final Map<Long, User> users = new HashMap<>();
+    private final Set<String> logins = new HashSet<>();
+    private final Set<String> emails = new HashSet<>();
```
Супер с отдельной коллекций, чтобы каждый раз не проверять всю коллецию пользователей, здорово, что проверяете что email есть или нет

Commit:
Final project num 11 done
Added service logic and new functionality
All postman tests passed
Added unit tests
Corrected based on recommendations

Commit:
Final project num 11 done
Added service logic and new functionality
All postman tests passed
Added unit tests
Corrected based on recommendations
